File: app/src/main/java/com/stark/mypratice/viewevent/custom_view/CircleView.java
package com.stark.mypratice.viewevent.custom_view;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.util.AttributeSet;
import android.view.View;

public class CircleView extends View{

    private Paint mPaint = new Paint(Paint.ANTI_ALIAS_FLAG);

    public CircleView(Context context) {
        this(context, null);
    }

    public CircleView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public CircleView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);

        init();
    }

    /**
     * 初始化
     */
    private void init() {
        mPaint.setColor(Color.RED);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        int width = getWidth();
        int height = getHeight();
        int radius = Math.min(width, height) / 2;
//        canvas.drawCircle(width/2, height/2, width > height ? height/2 : width/2, mPaint);
        canvas.drawCircle(width/2, height/2, radius, mPaint);
    }
}

File: app/src/main/java/com/stark/mypratice/view/draw/CircleView.kt
/**
 * Copyright (C), 2007-2021, 未来穿戴有限公司
 * FileName: CircleView
 * Author: lyl
 * Date: 2021/10/26 11:04
 * Description: 用一句话描述下
 */
package com.stark.mypratice.view.draw

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.util.AttributeSet
import android.util.Log
import android.view.View
import com.stark.mypratice.dp

/**
 * @ProjectName: MyPratice
 * @Package: com.stark.mypratice.view.draw
 * @ClassName: CircleView
 * @Description: 用一句话描述下
 * @Author: lpq
 * @CreateDate: 2021/10/26 11:04
 */
class CircleView(context: Context?, attrs: AttributeSet?) : View(context, attrs) {
    private val mPaint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
        color = Color.CYAN
    }
    private var radius = 10.dp
        set(value) {
            field = value
            invalidate()
        }

    override fun onDraw(canvas: Canvas) {
        canvas.drawCircle(width/2f, height/2f, radius, mPaint)
    }
}